homeassistant/components/airly/sensor.py

```diff
@@ -112,6 +116,34 @@ class AirlySensorEntityDescription(SensorEntityDescription):
         state_class=SensorStateClass.MEASUREMENT,
         value=lambda value: round(value, 1),
     ),
+    AirlySensorEntityDescription(
+        key=ATTR_API_CO,
+        device_class=SensorDeviceClass.CO,
+        name=ATTR_API_CO,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
```

Comment:
This device class should have unit ppm.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Thanks for the review :smiley: 

I think the usual device class usage of `carbon_monoxide` is in the context of indoor personal exposure measurement (e.g. smart CO/fire alarms), in which case `ppm` measurements make sense.

However outdoor air quality measurement of CO and other harmful gases are overwhelmingly reported in concentrations of `µg/m³`, as is the case with this Airly integration.

IMHO we should expand the device class to encompass both `ppm` and `µg/m³` units.
If you agree I'll gladly put this change into a new PR.

Reply:
Just had a quick look at `UnitSystem` and understand that this change is a bit more complex than just modifying the `carbon_monoxide` device class.

My approach would be to create a `CONCENTRATION_UNITS` or similar consisting of `ppm`, `ppb`, and `µg/m³`. This could be used by the various device classes that measure concentrations of gases or particulate matter.

Reply:
Changing the unit of a device class or allowing the user to change it, requires an architecture discussion. It's ok to start one, but first we should address the issue here and remove the device class where we've used an incorrect unit.

Reply:
Understood, thanks for the guidance on this.

I'll create a PR with some conversion logic in the Airly CO sensor so it'll report `ppm` correctly with the updated device class unit.

I'll also have a deeper look at the UnitSystem design and put together an architecture discussion, after sleeping on it I realized that it can't be a straightforward unit to unit conversion as the calculation between `ppm` and `µg/m³` varies for each chemical compound due to different molecular masses.

Reply:
See requested change in PR #78649